File: JAVA/PRACTICE/Java/SelectionSort.java
package Java;

class SelectionSort {

    static void sortNumbers(int numbers[]){

        for(int i = 0 ; i < numbers.length ; i ++){

            int minimum = i;

            for(int j = i + 1; j < numbers.length ; j ++){

                if(numbers[j] < numbers[minimum]){
                    minimum = j;
                }
            }  
        
            int temp =  numbers[minimum];
            numbers[minimum] = numbers[i];
            numbers[i] = temp;

        }       
        
        for(int i =0; i<numbers.length; i++){
            System.out.print(numbers[i] +" ");
        }

    }

    public static void main(String args[]){

        int numbers[] = {5,1,4,2,8,9};

        sortNumbers(numbers);
    }
}

File: JAVASCRIPT/PRACTICE/Js/SelectionSort.js
class SelectionSort{

    sortNumbers = (numbers) =>{

        for(let i =0; i < numbers.length ; i++){
            let min = i;
            for(let j = i + 1; j < numbers.length; j ++){
                if(numbers[j] < numbers[min]){
                    min = j;
                }
            }
            
            [numbers[i], numbers[min] ] = [ numbers[min], numbers[i]];

        }

        console.log(numbers);

    };
}
const selectionSort = new SelectionSort();
selectionSort.sortNumbers([10,7,3,5,8,11,4,2])